### PR TITLE
Clean up variables between provider and client

### PR DESCRIPTION
**Pull request by ilausuch**
In the code we have extra complexity in order to flexibility that
we don't need.

This PR clean up the duplicated variables between provider and clients

- Related ticket: https://progress.opensuse.org/issues/103386
-Verification run:
http://openqa.suse.de/t8001314 (AWS) OK
http://openqa.suse.de/tests/8005346 (EKS) OK
https://openqa.suse.de/tests/8001313 (Azure) OK
https://openqa.suse.de/tests/8001313 (Google) OK

### CODE REVIEW

File: lib/publiccloud/gce.pm [hunk 1]
```diff
@@ -30,15 +30,7 @@ has username => sub { get_var('PUBLIC_CLOUD_USER', 'susetest') };
 sub init {
     my ($self, %params) = @_;
     $self->SUPER::init();
-    $self->provider_client(publiccloud::gcp_client->new(
-            region => $self->region,
```
Comment (jlausuch) [inline lib/publiccloud/gce.pm:34]: `region` is not defined in `gcp_client.pm`. Can we remove it from here?

Comment (ilausuch) [reply lib/publiccloud/gce.pm:34]: Take a look to lib/publiccloud/aws_client.pm
`has region => sub { get_var('PUBLIC_CLOUD_REGION', 'eu-central-1') };

Comment (jlausuch) [reply lib/publiccloud/gce.pm:34]: yes, I mean in `gcp_client.pm`. Maybe, it's just ok and `region` is not needed in GCP?

**Comment by asmorodskyi** [comment]
it is certainly an improvement and move in a right direction . but just to keep things clear our end goal is to remove duplicates . here we just breaking coupling but still keeping duplicates . just an example tentant_id is currently presented in azure and in azure client

File: lib/publiccloud/gce.pm (hunk 1 above)
Comment (ilausuch) [reply lib/publiccloud/gce.pm:34]: Ah ok, sorry
it uses
`has gcr_zone => sub { get_var('PUBLIC_CLOUD_GCR_ZONE', 'eu.gcr.io') };

**Comment by ilausuch** [comment]
@asmorodskyi Ok, understook, let me improve that

**Review by grisu48 — APPROVED** [review]
LGTM

File: lib/publiccloud/azure.pm [hunk 2]
```diff
@@ -16,26 +16,16 @@ use testapi qw(is_serial_terminal :DEFAULT);
...
-has tenantid => sub { get_var('PUBLIC_CLOUD_AZURE_TENANT_ID') };
-has subscription => sub { get_var('PUBLIC_CLOUD_AZURE_SUBSCRIPTION_ID') };
-has region => sub { get_var('PUBLIC_CLOUD_REGION', 'westeurope') };
-has username => sub { get_var('PUBLIC_CLOUD_USER', 'azureuser') };
 has resource_group => 'openqa-upload';
 has storage_account => 'openqa';
 has container => 'sle-images';
 has lease_id => undef;
 has vault => undef;
-has provider_client => undef;
```
Comment (asmorodskyi) [inline lib/publiccloud/azure.pm:28]: is it correct ? we still using it , right ?

Comment (asmorodskyi) [reply lib/publiccloud/azure.pm:28]: ok , you moved it to provider . nice !

**Review by asmorodskyi — APPROVED** [review]
LGTM

File: lib/publiccloud/azure.pm (hunk 2 above)
Comment (ilausuch) [reply lib/publiccloud/azure.pm:28]: yes :)

**Review by jlausuch — APPROVED** [review]

**Pull request merged**